providers/hexonet/domains.go

```diff
@@ -24,3 +26,39 @@ func (n *HXClient) EnsureDomainExists(domain string) error {
 	}
 	return nil
 }
+
+// ListZones lists all the
+func (n *HXClient) ListZones() ([]string, error) {
+
+	// Basic
+
+	r := n.client.Request(map[string]interface{}{
```

Comment:
nearly :-)

the main "problem" is that our API limits the rows returned by list queries by default (in case parameter LIMIT is not set). This is for security and performance reasons.

Check this method `RequestAllResponsePages` instead: https://github.com/hexonet/go-sdk/blob/master/apiclient/apiclient.go#L437

```go
r := n.client.RequestAllResponsePages(map[string]interface{}{
	"COMMAND": "QueryDNSZoneList",
        "FIRST": "0",
        "LIMIT": "1000"
})
//now you have an array of API responses available to loop over.
```

This also applies to the below Premium DNS query accordingly.